docs/source/development.md

```diff
@@ -40,3 +40,25 @@ pixi shell --environment dev
 The environment `dev` contains the most critical development dependencies. However, you can view the available environment sets in the `pixi.toml` to start a different environment if you would like. For example, if you are only updating the docs, you can do this by setting the environment flag to `docs`.
 
 While in the `pixi` shell and/or Docker container, all changes made in the repository or to the `~/.bashrc` file persist in your local file system. You can also use `git` with no issue to push changes.
+
+## Building and Running C++ Code
+We can easily run C++ code using `Pixi`.
+From within the `dev` enviroment, the available commands are:
+- `cpp-ctest` will run all tests registered with `CTest`. This is the command used in the CI to verify unit tests, so every tests should be registered with CTest - see [below](#testing).
+- `cpp-test-dummy` will run just the dummy tests and will do so using the Catch2 framework. This command will be removed in the future and either replaced with commands to run specific tests or replaced with nothing at all.
+- `cmake` which will re-build the cmake.
+- `cpp-build` which will compile the code.
+
+Note that all the commands will automatically run the commands they depend on. For this reason to re-build the cmake, compile the code, and run the tests, all we need to do is run `cpp-ctest`.
```

Comment:
I think we can probably remove detail like this in the future once we explain how `pixi` works elsewhere in the docs. For now, it's probably fine, but we should have a separate PR where we streamline the docs down the road.